src/components/SwapTokenInput.tsx

```diff
@@ -0,0 +1,289 @@
+import * as React from "react"
+
+import {
+  Button,
+  Chip,
+  InputBase,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material"
+import { ReactElement, useRef, useState } from "react"
+import { SWAP_TYPES, TOKENS_MAP } from "../constants"
+import { commify, formatBNToString } from "../utils"
+import { styled, useTheme } from "@mui/material/styles"
+import { ArrowDropDown } from "@mui/icons-material"
+import Autocomplete from "@mui/material/Autocomplete"
+import { BigNumber } from "ethers"
+import Box from "@mui/material/Box"
+import ClickAwayListener from "@mui/material/ClickAwayListener"
+import Popper from "@mui/material/Popper"
+import Search from "@mui/icons-material/Search"
+import { TokenOption } from "../pages/Swap"
+import { useTranslation } from "react-i18next"
+
+const StyledPopper = styled(Popper)(({ theme }) => ({
+  border: `1px solid ${theme.palette.other.divider}`,
+  borderRadius: theme.shape.borderRadius,
+  width: 400,
+  padding: theme.spacing(2),
+  zIndex: theme.zIndex.modal,
+  backgroundColor: theme.palette.background.paper,
+}))
+
+interface SwapTokenInputProps {
+  tokens: TokenOption[]
+  selected?: string
+  inputValue: string
+  inputValueUSD: BigNumber
+  isSwapFrom?: boolean
```

Comment:
This is currently unused in your implementation. In the existing implementation, it is used to disable the user from selecting / editing the `to` input field